python/vyos/ifconfig/erspan.py

```diff
@@ -0,0 +1,192 @@
+class _ERSpan(Interface):
+    """
+    _ERSpan: private base class for ERSPAN tunnels
+    """
+    default = {
+        **Interface.default,
+        **{
+            'type': 'erspan',
+        }
+    }
+    definition = {
+        **Interface.definition,
+        **{
+            'section': 'erspan',
+            'prefixes': ['ersp',],
+        },
+    }
+    
+    options = ['local_ip','remote_ip','encapsulation','parameters']
+    
+    def __init__(self,ifname,**config):
+        self.config = deepcopy(config) if config else {}
+        super().__init__(ifname, **self.config)
+        
+    
+    def update(self, config):
+        
+        # Enable/Disable of an interface must always be done at the end of the
+        # derived class to make use of the ref-counting set_admin_state()
+        # function. We will only enable the interface if 'up' was called as
+        # often as 'down'. This is required by some interface implementations
+        # as certain parameters can only be changed when the interface is
+        # in admin-down state. This ensures the link does not flap during
+        # reconfiguration.
+        super().update(config)
+        state = 'down' if 'disable' in config else 'up'
+        self.set_admin_state(state)
+    
+    def _create(self):
+        pass
+
+class ERSpanIf(_ERSpan):
+    """
+    ERSpanIf: private base class for ERSPAN Over GRE and IPv4 tunnels
+    """
+    
+    def _create(self):
+        ifname = self.config['ifname']
+        local_ip = self.config['local_ip']
+        remote_ip = self.config['remote_ip']
+        key = self.config['parameters']['ip']['key']
+        version = self.config['parameters']['version']
+        command = f'ip link add dev {ifname} type erspan local {local_ip} remote {remote_ip} seq key {key} erspan_ver {version}'
+        
+        if int(version) == 1:
+            idx=dict_search('parameters.erspan.idx',self.config)
+            if idx:
+                command += f' erspan {idx}'
+        elif int(version) == 2:
+            direction=dict_search('parameters.erspan.direction',self.config)
+            if direction:
+                command += f' erspan_dir {direction}'
+            hwid=dict_search('parameters.erspan.hwid',self.config)
+            if hwid:
+                command += f' erspan_hwid {hwid}'
+        
+        ttl = dict_search('parameters.ip.ttl',self.config)
+        if ttl:
+            command += f' ttl {ttl}'
+        tos = dict_search('parameters.ip.tos',self.config)
+        if tos:
+            command += f' tos {tos}'
+                
+        self._cmd(command)
+    
+    def update(self, config):
+        ifname = config['ifname']
+        local_ip = config['local_ip']
+        remote_ip = config['remote_ip']
+        key = config['parameters']['ip']['key']
+        version = config['parameters']['version']
+        command = f'ip link set dev {ifname} type erspan local {local_ip} remote {remote_ip} seq key {key} erspan_ver {version}'
```

Comment:
As already mentioned, we do not change low level interface parameters in `update()`. See `tunnel.py:change_options()` and `tunnel.py:_create()`.

Any reason this is not derived from the `_Tunnel()` class`

Reply:
Please see if my latest code structure is what you expect